--- a/net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
@@ -6,9 +_,22 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.DispenserBlock;
+import org.bukkit.block.Block;
+import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+import org.bukkit.craftbukkit.util.CraftVector;
+import org.bukkit.event.block.BlockDispenseEvent;
 
 public class DefaultDispenseItemBehavior implements DispenseItemBehavior {
     private static final int DEFAULT_ACCURACY = 6;
+    // CraftBukkit start
+    private boolean dropper;
+
+    public DefaultDispenseItemBehavior(boolean dropper) {
+        this.dropper = dropper;
+    }
+    // CraftBukkit end
+    public DefaultDispenseItemBehavior() {}
 
     @Override
     public final ItemStack dispense(BlockSource p_302432_, ItemStack p_123392_) {
@@ -22,10 +_,35 @@
         Direction direction = p_302420_.state().getValue(DispenserBlock.FACING);
         Position position = DispenserBlock.getDispensePosition(p_302420_);
         ItemStack itemstack = p_123386_.split(1);
-        spawnItem(p_302420_.level(), itemstack, 6, direction, position);
+        // CraftBukkit start
+        if (!spawnItem(p_302420_.level(), itemstack, 6, direction, p_302420_, dropper)) {
+            itemstack.grow(1);
+        }
+        // CraftBukkit end
         return p_123386_;
     }
 
+    public static ItemEntity prepareItem(Level p_123379_, ItemStack p_123380_, int p_123381_, Direction p_123382_, Position p_123383_) {
+        double d0 = p_123383_.x();
+        double d1 = p_123383_.y();
+        double d2 = p_123383_.z();
+        if (p_123382_.getAxis() == Direction.Axis.Y) {
+            d1 -= 0.125;
+        } else {
+            d1 -= 0.15625;
+        }
+
+        ItemEntity itementity = new ItemEntity(p_123379_, d0, d1, d2, p_123380_);
+        double d3 = p_123379_.random.nextDouble() * 0.1 + 0.2;
+        itementity.setDeltaMovement(
+                p_123379_.random.triangle((double)p_123382_.getStepX() * d3, 0.0172275 * (double)p_123381_),
+                p_123379_.random.triangle(0.2, 0.0172275 * (double)p_123381_),
+                p_123379_.random.triangle((double)p_123382_.getStepZ() * d3, 0.0172275 * (double)p_123381_)
+        );
+        // CraftBukkit start
+        return itementity;
+    }
+
     public static void spawnItem(Level p_123379_, ItemStack p_123380_, int p_123381_, Direction p_123382_, Position p_123383_) {
         double d0 = p_123383_.x();
         double d1 = p_123383_.y();
@@ -39,11 +_,49 @@
         ItemEntity itementity = new ItemEntity(p_123379_, d0, d1, d2, p_123380_);
         double d3 = p_123379_.random.nextDouble() * 0.1 + 0.2;
         itementity.setDeltaMovement(
-            p_123379_.random.triangle((double)p_123382_.getStepX() * d3, 0.0172275 * (double)p_123381_),
-            p_123379_.random.triangle(0.2, 0.0172275 * (double)p_123381_),
-            p_123379_.random.triangle((double)p_123382_.getStepZ() * d3, 0.0172275 * (double)p_123381_)
+                p_123379_.random.triangle((double)p_123382_.getStepX() * d3, 0.0172275 * (double)p_123381_),
+                p_123379_.random.triangle(0.2, 0.0172275 * (double)p_123381_),
+                p_123379_.random.triangle((double)p_123382_.getStepZ() * d3, 0.0172275 * (double)p_123381_)
         );
         p_123379_.addFreshEntity(itementity);
+    }
+
+    public static boolean spawnItem(Level p_123379_, ItemStack p_123380_, int p_123381_, Direction p_123382_, BlockSource sourceblock, boolean dropper) {
+        if (p_123380_.isEmpty()) return true;
+        Position iposition = DispenserBlock.getDispensePosition(sourceblock);
+        ItemEntity entityitem = prepareItem(p_123379_, p_123380_, p_123381_, p_123382_, iposition);
+
+        Block block = CraftBlock.at(p_123379_, sourceblock.pos());
+        CraftItemStack craftItem = CraftItemStack.asCraftMirror(p_123380_);
+
+        BlockDispenseEvent event = new BlockDispenseEvent(block, craftItem.clone(), CraftVector.toBukkit(entityitem.getDeltaMovement()));
+        if (!DispenserBlock.eventFired) {
+            p_123379_.getCraftServer().getPluginManager().callEvent(event);
+        }
+
+        if (event.isCancelled()) {
+            return false;
+        }
+
+        entityitem.setItem(CraftItemStack.asNMSCopy(event.getItem()));
+        entityitem.setDeltaMovement(CraftVector.toNMS(event.getVelocity()));
+
+        if (!dropper && !event.getItem().getType().equals(craftItem.getType())) {
+            // Chain to handler for new item
+            ItemStack eventStack = CraftItemStack.asNMSCopy(event.getItem());
+            DispenseItemBehavior idispensebehavior = (DispenseItemBehavior) DispenserBlock.DISPENSER_REGISTRY.get(eventStack.getItem());
+            if (idispensebehavior != DispenseItemBehavior.NOOP && idispensebehavior.getClass() != DefaultDispenseItemBehavior.class) {
+                idispensebehavior.dispense(sourceblock, eventStack);
+            } else {
+                p_123379_.addFreshEntity(entityitem);
+            }
+            return false;
+        }
+
+        p_123379_.addFreshEntity(entityitem);
+
+        return true;
+        // CraftBukkit end
     }
 
     protected void playSound(BlockSource p_302471_) {
