--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -9,11 +_,15 @@
 import java.util.Objects;
 import javax.annotation.Nullable;
 import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.util.ClassTreeIdRegistry;
+import net.minecraft.world.entity.Entity;
 import org.apache.commons.lang3.ObjectUtils;
 import org.slf4j.Logger;
+import org.taiyitistmc.injection.network.syncher.InjectionSynchedEntityData;
 
-public class SynchedEntityData {
+public class SynchedEntityData implements InjectionSynchedEntityData {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int MAX_ID_VALUE = 254;
     static final ClassTreeIdRegistry ID_REGISTRY = new ClassTreeIdRegistry();
@@ -26,12 +_,30 @@
         this.itemsById = p_326032_;
     }
 
+    @Override
+    // CraftBukkit start - add method from above
+    public <T> void markDirty(EntityDataAccessor<T> datawatcherobject) {
+        this.getItem(datawatcherobject).setDirty(true);
+        this.isDirty = true;
+    }
+    // CraftBukkit end
+
+    @Override
+    public void refresh(ServerPlayer player) {
+        var list = this.getNonDefaultValues();
+        if (list != null && this.entity instanceof Entity entity) {
+            player.connection.send(new ClientboundSetEntityDataPacket(entity.getId(), list));
+        }
+    }
+
     public static <T> EntityDataAccessor<T> defineId(Class<? extends SyncedDataHolder> p_135354_, EntityDataSerializer<T> p_135355_) {
-        if (LOGGER.isDebugEnabled()) {
+        if (true || LOGGER.isDebugEnabled()) { // Forge: This is very useful for mods that register keys on classes that are not their own
             try {
                 Class<?> oclass = Class.forName(Thread.currentThread().getStackTrace()[2].getClassName());
                 if (!oclass.equals(p_135354_)) {
-                    LOGGER.debug("defineId called for: {} from {}", p_135354_, oclass, new RuntimeException());
+                    // Forge: log at warn, mods should not add to classes that they don't own, and only add stacktrace when in debug is enabled as it is mostly not needed and consumes time
+                    if (LOGGER.isDebugEnabled()) LOGGER.warn("defineId called for: {} from {}", p_135354_, oclass, new RuntimeException());
+                    else LOGGER.warn("defineId called for: {} from {}", p_135354_, oclass);
                 }
             } catch (ClassNotFoundException classnotfoundexception) {
             }
