--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
@@ -20,7 +_,6 @@
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.nio.file.attribute.BasicFileAttributes;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
@@ -48,8 +_,9 @@
 import net.minecraft.world.level.storage.LevelStorageSource;
 import org.apache.commons.io.IOUtils;
 import org.slf4j.Logger;
+import org.celestial_artistry.neotenet.injection.world.level.levelgen.structure.templatesystem.InjectionStructureTemplateManager;
 
-public class StructureTemplateManager {
+public class StructureTemplateManager implements InjectionStructureTemplateManager {
     private static final Logger LOGGER = LogUtils.getLogger();
     public static final String STRUCTURE_RESOURCE_DIRECTORY_NAME = "structure";
     private static final String STRUCTURE_GENERATED_DIRECTORY_NAME = "structures";
