--- a/net/minecraft/world/level/entity/EntityAccess.java
+++ b/net/minecraft/world/level/entity/EntityAccess.java
@@ -5,8 +_,9 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.phys.AABB;
+import org.teneted.neotenet.injection.world.level.entity.InjectionEntityAccess;
 
-public interface EntityAccess {
+public interface EntityAccess extends InjectionEntityAccess {
     int getId();
 
     UUID getUUID();
