--- a/net/minecraft/world/level/BaseCommandBlock.java
+++ b/net/minecraft/world/level/BaseCommandBlock.java
@@ -129,7 +_,7 @@
         return true;
     }
 
-    public BaseCommandBlock.@Nullable CloseableCommandBlockSource createSource(ServerLevel level) {
+    private BaseCommandBlock.@Nullable CloseableCommandBlockSource createSource(ServerLevel level) {
         return this.trackOutput ? new BaseCommandBlock.CloseableCommandBlockSource(level) : null;
     }
 
