--- a/net/minecraft/world/level/ServerLevelAccessor.java
+++ b/net/minecraft/world/level/ServerLevelAccessor.java
@@ -2,8 +_,9 @@
 
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.Entity;
+import org.celestial_artistry.neotenet.injection.world.level.InjectionServerLevelAccessor;
 
-public interface ServerLevelAccessor extends LevelAccessor {
+public interface ServerLevelAccessor extends LevelAccessor, InjectionServerLevelAccessor {
     ServerLevel getLevel();
 
     default void addFreshEntityWithPassengers(Entity p_47206_) {
