--- a/net/minecraft/world/level/ServerLevelAccessor.java
+++ b/net/minecraft/world/level/ServerLevelAccessor.java
@@ -4,8 +_,9 @@
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.DifficultyInstance;
 import net.minecraft.world.entity.Entity;
+import org.teneted.neotenet.injection.world.level.ServerLevelAccessorInjection;
 
-public interface ServerLevelAccessor extends LevelAccessor {
+public interface ServerLevelAccessor extends LevelAccessor, ServerLevelAccessorInjection {
     ServerLevel getLevel();
 
     DifficultyInstance getCurrentDifficultyAt(BlockPos pos);
