--- a/net/minecraft/world/entity/animal/rabbit/Rabbit.java
+++ b/net/minecraft/world/entity/animal/rabbit/Rabbit.java
@@ -116,7 +_,7 @@
     }
 
     @Override
-    public void registerGoals() {
+    protected void registerGoals() {
         this.goalSelector.addGoal(1, new FloatGoal(this));
         this.goalSelector.addGoal(1, new ClimbOnTopOfPowderSnowGoal(this, this.level()));
         this.goalSelector.addGoal(1, new Rabbit.RabbitPanicGoal(this, 2.2));
@@ -365,7 +_,7 @@
         return Rabbit.Variant.byId(this.entityData.get(DATA_TYPE_ID));
     }
 
-    public void setVariant(Rabbit.Variant variant) {
+    private void setVariant(Rabbit.Variant variant) {
         if (variant == Rabbit.Variant.EVIL) {
             this.getAttribute(Attributes.ARMOR).setBaseValue(8.0);
             this.goalSelector.addGoal(4, new MeleeAttackGoal(this, 1.4, true));
@@ -599,7 +_,7 @@
         @Override
         public boolean canUse() {
             if (this.nextStartTick <= 0) {
-                if (!getServerLevel(this.rabbit).getGameRules().get(GameRules.MOB_GRIEFING)) {
+                if (!net.neoforged.neoforge.event.EventHooks.canEntityGrief(getServerLevel(this.rabbit.level()), this.rabbit)) {
                     return false;
                 }
 
