--- a/net/minecraft/world/entity/animal/sniffer/Sniffer.java
+++ b/net/minecraft/world/entity/animal/sniffer/Sniffer.java
@@ -159,7 +_,7 @@
         return Leashable.createQuadLeashOffsets(this, -0.01, 0.63, 0.38, 1.15);
     }
 
-    public Sniffer.State getState() {
+    private Sniffer.State getState() {
         return this.entityData.get(DATA_STATE);
     }
 
@@ -255,7 +_,7 @@
         return this;
     }
 
-    public Optional<BlockPos> calculateDigPosition() {
+    Optional<BlockPos> calculateDigPosition() {
         return IntStream.range(0, 5)
             .mapToObj(idx -> LandRandomPos.getPos(this, 10 + 2 * idx, 3))
             .filter(Objects::nonNull)
@@ -266,7 +_,7 @@
             .findFirst();
     }
 
-    public boolean canDig() {
+    boolean canDig() {
         return !this.isPanicking()
             && !this.isTempted()
             && !this.isBaby()
@@ -308,7 +_,7 @@
                 if (this.tickCount % 10 == 0) {
                     this.level()
                         .playLocalSound(
-                            this.getX(), this.getY(), this.getZ(), stateBelow.getSoundType().getHitSound(), this.getSoundSource(), 0.5F, 0.5F, false
+                            this.getX(), this.getY(), this.getZ(), stateBelow.getSoundType(level(), head.below(), this).getHitSound(), this.getSoundSource(), 0.5F, 0.5F, false
                         );
                 }
             }
@@ -321,14 +_,14 @@
         return this;
     }
 
-    public Sniffer storeExploredPosition(BlockPos position) {
+    private Sniffer storeExploredPosition(BlockPos position) {
         List<GlobalPos> updated = this.getExploredPositions().limit(20L).collect(Collectors.toList());
         updated.add(0, GlobalPos.of(this.level().dimension(), position));
         this.getBrain().setMemory(MemoryModuleType.SNIFFER_EXPLORED_POSITIONS, updated);
         return this;
     }
 
-    public Stream<GlobalPos> getExploredPositions() {
+    private Stream<GlobalPos> getExploredPositions() {
         return this.getBrain().getMemory(MemoryModuleType.SNIFFER_EXPLORED_POSITIONS).stream().flatMap(Collection::stream);
     }
 
