--- a/net/minecraft/world/item/trading/Merchant.java
+++ b/net/minecraft/world/item/trading/Merchant.java
@@ -5,12 +_,12 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.world.SimpleMenuProvider;
-import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.MerchantMenu;
 import net.minecraft.world.item.ItemStack;
+import org.celestial_artistry.neotenet.injection.world.item.trading.InjectionMerchant;
 
-public interface Merchant {
+public interface Merchant extends InjectionMerchant {
     void setTradingPlayer(@Nullable Player p_45307_);
 
     @Nullable
