--- a/net/minecraft/server/dedicated/Settings.java
+++ b/net/minecraft/server/dedicated/Settings.java
@@ -25,7 +_,7 @@
 
 public abstract class Settings<T extends Settings<T>> {
     private static final Logger LOGGER = LogUtils.getLogger();
-    public final Properties properties;
+    protected final Properties properties;
 
     public Settings(Properties properties) {
         this.properties = properties;
@@ -58,7 +_,7 @@
 
     public void store(Path output) {
         try (Writer os = Files.newBufferedWriter(output, StandardCharsets.UTF_8)) {
-            this.properties.store(os, "Minecraft server properties");
+            net.neoforged.neoforge.common.util.SortedProperties.store(this.properties, os, "Minecraft server properties");
         } catch (IOException e) {
             LOGGER.error("Failed to store properties to file: {}", output);
         }
