--- a/net/minecraft/world/entity/Display.java
+++ b/net/minecraft/world/entity/Display.java
@@ -49,7 +_,7 @@
     private static final EntityDataAccessor<Integer> DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
         Display.class, EntityDataSerializers.INT
     );
-    public static final EntityDataAccessor<Integer> DATA_POS_ROT_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
+    private static final EntityDataAccessor<Integer> DATA_POS_ROT_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
         Display.class, EntityDataSerializers.INT
     );
     private static final EntityDataAccessor<Vector3fc> DATA_TRANSLATION_ID = SynchedEntityData.defineId(Display.class, EntityDataSerializers.VECTOR3);
@@ -141,7 +_,7 @@
         return false;
     }
 
-    public static Transformation createTransformation(SynchedEntityData entityData) {
+    private static Transformation createTransformation(SynchedEntityData entityData) {
         Vector3fc translation = entityData.get(DATA_TRANSLATION_ID);
         Quaternionfc leftRotation = entityData.get(DATA_LEFT_ROTATION_ID);
         Vector3fc scale = entityData.get(DATA_SCALE_ID);
@@ -227,7 +_,7 @@
         this.setBrightnessOverride(input.read("brightness", Brightness.CODEC).orElse(null));
     }
 
-    public void setTransformation(Transformation transformation) {
+    private void setTransformation(Transformation transformation) {
         this.entityData.set(DATA_TRANSLATION_ID, transformation.translation());
         this.entityData.set(DATA_LEFT_ROTATION_ID, transformation.leftRotation());
         this.entityData.set(DATA_SCALE_ID, transformation.scale());
@@ -271,19 +_,19 @@
         return this.renderState;
     }
 
-    public void setTransformationInterpolationDuration(int duration) {
+    private void setTransformationInterpolationDuration(int duration) {
         this.entityData.set(DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID, duration);
     }
 
-    public int getTransformationInterpolationDuration() {
+    private int getTransformationInterpolationDuration() {
         return this.entityData.get(DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID);
     }
 
-    public void setTransformationInterpolationDelay(int ticks) {
+    private void setTransformationInterpolationDelay(int ticks) {
         this.entityData.set(DATA_TRANSFORMATION_INTERPOLATION_START_DELTA_TICKS_ID, ticks, true);
     }
 
-    public int getTransformationInterpolationDelay() {
+    private int getTransformationInterpolationDelay() {
         return this.entityData.get(DATA_TRANSFORMATION_INTERPOLATION_START_DELTA_TICKS_ID);
     }
 
@@ -295,19 +_,19 @@
         return this.entityData.get(DATA_POS_ROT_INTERPOLATION_DURATION_ID);
     }
 
-    public void setBillboardConstraints(Display.BillboardConstraints constraints) {
+    private void setBillboardConstraints(Display.BillboardConstraints constraints) {
         this.entityData.set(DATA_BILLBOARD_RENDER_CONSTRAINTS_ID, constraints.getId());
     }
 
-    public Display.BillboardConstraints getBillboardConstraints() {
+    private Display.BillboardConstraints getBillboardConstraints() {
         return Display.BillboardConstraints.BY_ID.apply(this.entityData.get(DATA_BILLBOARD_RENDER_CONSTRAINTS_ID));
     }
 
-    public void setBrightnessOverride(@Nullable Brightness brightness) {
+    private void setBrightnessOverride(@Nullable Brightness brightness) {
         this.entityData.set(DATA_BRIGHTNESS_OVERRIDE_ID, brightness != null ? brightness.pack() : -1);
     }
 
-    public @Nullable Brightness getBrightnessOverride() {
+    private @Nullable Brightness getBrightnessOverride() {
         int value = this.entityData.get(DATA_BRIGHTNESS_OVERRIDE_ID);
         return value != -1 ? Brightness.unpack(value) : null;
     }
@@ -316,47 +_,47 @@
         return this.entityData.get(DATA_BRIGHTNESS_OVERRIDE_ID);
     }
 
-    public void setViewRange(float range) {
+    private void setViewRange(float range) {
         this.entityData.set(DATA_VIEW_RANGE_ID, range);
     }
 
-    public float getViewRange() {
+    private float getViewRange() {
         return this.entityData.get(DATA_VIEW_RANGE_ID);
     }
 
-    public void setShadowRadius(float size) {
+    private void setShadowRadius(float size) {
         this.entityData.set(DATA_SHADOW_RADIUS_ID, size);
     }
 
-    public float getShadowRadius() {
+    private float getShadowRadius() {
         return this.entityData.get(DATA_SHADOW_RADIUS_ID);
     }
 
-    public void setShadowStrength(float strength) {
+    private void setShadowStrength(float strength) {
         this.entityData.set(DATA_SHADOW_STRENGTH_ID, strength);
     }
 
-    public float getShadowStrength() {
+    private float getShadowStrength() {
         return this.entityData.get(DATA_SHADOW_STRENGTH_ID);
     }
 
-    public void setWidth(float width) {
+    private void setWidth(float width) {
         this.entityData.set(DATA_WIDTH_ID, width);
     }
 
-    public float getWidth() {
+    private float getWidth() {
         return this.entityData.get(DATA_WIDTH_ID);
     }
 
-    public void setHeight(float width) {
+    private void setHeight(float width) {
         this.entityData.set(DATA_HEIGHT_ID, width);
     }
 
-    public int getGlowColorOverride() {
+    private int getGlowColorOverride() {
         return this.entityData.get(DATA_GLOW_COLOR_OVERRIDE_ID);
     }
 
-    public void setGlowColorOverride(int value) {
+    private void setGlowColorOverride(int value) {
         this.entityData.set(DATA_GLOW_COLOR_OVERRIDE_ID, value);
     }
 
@@ -373,7 +_,7 @@
         return result;
     }
 
-    public float getHeight() {
+    private float getHeight() {
         return this.entityData.get(DATA_HEIGHT_ID);
     }
 
@@ -483,11 +_,11 @@
             }
         }
 
-        public BlockState getBlockState() {
+        private BlockState getBlockState() {
             return this.entityData.get(DATA_BLOCK_STATE_ID);
         }
 
-        public void setBlockState(BlockState blockState) {
+        private void setBlockState(BlockState blockState) {
             this.entityData.set(DATA_BLOCK_STATE_ID, blockState);
         }
 
@@ -579,19 +_,19 @@
             }
         }
 
-        public ItemStack getItemStack() {
+        private ItemStack getItemStack() {
             return this.entityData.get(DATA_ITEM_STACK_ID);
         }
 
-        public void setItemStack(ItemStack item) {
+        private void setItemStack(ItemStack item) {
             this.entityData.set(DATA_ITEM_STACK_ID, item);
         }
 
-        public void setItemTransform(ItemDisplayContext transform) {
+        private void setItemTransform(ItemDisplayContext transform) {
             this.entityData.set(DATA_ITEM_DISPLAY_ID, transform.getId());
         }
 
-        public ItemDisplayContext getItemTransform() {
+        private ItemDisplayContext getItemTransform() {
             return ItemDisplayContext.BY_ID.apply(this.entityData.get(DATA_ITEM_DISPLAY_ID));
         }
 
@@ -673,8 +_,8 @@
         public static final int INITIAL_BACKGROUND = 1073741824;
         private static final int INITIAL_LINE_WIDTH = 200;
         private static final EntityDataAccessor<Component> DATA_TEXT_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.COMPONENT);
-        public static final EntityDataAccessor<Integer> DATA_LINE_WIDTH_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.INT);
-        public static final EntityDataAccessor<Integer> DATA_BACKGROUND_COLOR_ID = SynchedEntityData.defineId(
+        private static final EntityDataAccessor<Integer> DATA_LINE_WIDTH_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.INT);
+        private static final EntityDataAccessor<Integer> DATA_BACKGROUND_COLOR_ID = SynchedEntityData.defineId(
             Display.TextDisplay.class, EntityDataSerializers.INT
         );
         private static final EntityDataAccessor<Byte> DATA_TEXT_OPACITY_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.BYTE);
@@ -707,15 +_,15 @@
             }
         }
 
-        public Component getText() {
+        private Component getText() {
             return this.entityData.get(DATA_TEXT_ID);
         }
 
-        public void setText(Component text) {
+        private void setText(Component text) {
             this.entityData.set(DATA_TEXT_ID, text);
         }
 
-        public int getLineWidth() {
+        private int getLineWidth() {
             return this.entityData.get(DATA_LINE_WIDTH_ID);
         }
 
@@ -723,15 +_,15 @@
             this.entityData.set(DATA_LINE_WIDTH_ID, width);
         }
 
-        public byte getTextOpacity() {
+        private byte getTextOpacity() {
             return this.entityData.get(DATA_TEXT_OPACITY_ID);
         }
 
-        public void setTextOpacity(byte opacity) {
+        private void setTextOpacity(byte opacity) {
             this.entityData.set(DATA_TEXT_OPACITY_ID, opacity);
         }
 
-        public int getBackgroundColor() {
+        private int getBackgroundColor() {
             return this.entityData.get(DATA_BACKGROUND_COLOR_ID);
         }
 
@@ -739,11 +_,11 @@
             this.entityData.set(DATA_BACKGROUND_COLOR_ID, color);
         }
 
-        public byte getFlags() {
+        private byte getFlags() {
             return this.entityData.get(DATA_STYLE_FLAGS_ID);
         }
 
-        public void setFlags(byte flags) {
+        private void setFlags(byte flags) {
             this.entityData.set(DATA_STYLE_FLAGS_ID, flags);
         }
 
