--- a/net/minecraft/world/entity/monster/hoglin/Hoglin.java
+++ b/net/minecraft/world/entity/monster/hoglin/Hoglin.java
@@ -68,8 +_,8 @@
     private static final boolean DEFAULT_CANNOT_BE_HUNTED = false;
     public static final int CONVERSION_TIME = 300;
     private int attackAnimationRemainingTicks;
-    public int timeInOverworld = 0;
-    public boolean cannotBeHunted = false;
+    private int timeInOverworld = 0;
+    private boolean cannotBeHunted = false;
     private static final Brain.Provider<Hoglin> BRAIN_PROVIDER = Brain.<Hoglin>provider(
         List.of(SensorType.NEAREST_LIVING_ENTITIES, SensorType.NEAREST_PLAYERS, SensorType.NEAREST_ADULT, SensorType.HOGLIN_SPECIFIC_SENSOR),
         var0 -> HoglinAi.getActivities()
@@ -148,7 +_,7 @@
         HoglinAi.updateActivity(this);
         if (this.isConverting()) {
             this.timeInOverworld++;
-            if (this.timeInOverworld > 300) {
+            if (this.timeInOverworld > 300 && net.neoforged.neoforge.event.EventHooks.canLivingConvert(this, EntityTypes.ZOGLIN, (timer) -> this.timeInOverworld = timer)) {
                 this.makeSound(SoundEvents.HOGLIN_CONVERTED_TO_ZOMBIFIED);
                 this.finishConversion();
             }
@@ -248,7 +_,10 @@
 
     private void finishConversion() {
         this.convertTo(
-            EntityTypes.ZOGLIN, ConversionParams.single(this, true, false), zoglin -> zoglin.addEffect(new MobEffectInstance(MobEffects.NAUSEA, 200, 0))
+            EntityTypes.ZOGLIN, ConversionParams.single(this, true, false), zoglin -> {
+                zoglin.addEffect(new MobEffectInstance(MobEffects.NAUSEA, 200, 0));
+                net.neoforged.neoforge.event.EventHooks.onLivingConvert(this, zoglin);
+            }
         );
     }
 
@@ -287,7 +_,7 @@
         this.getEntityData().set(DATA_IMMUNE_TO_ZOMBIFICATION, isImmuneToZombification);
     }
 
-    public boolean isImmuneToZombification() {
+    private boolean isImmuneToZombification() {
         return this.getEntityData().get(DATA_IMMUNE_TO_ZOMBIFICATION);
     }
 
