--- a/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
+++ b/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
@@ -17,8 +_,9 @@
 import net.minecraft.world.level.Level;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import org.taiyitistmc.injection.world.entity.projectile.InjectionAbstractHurtingProjectile;
 
-public abstract class AbstractHurtingProjectile extends Projectile {
+public abstract class AbstractHurtingProjectile extends Projectile implements InjectionAbstractHurtingProjectile {
     public static final double INITAL_ACCELERATION_POWER = 0.1;
     public static final double DEFLECTION_SCALE = 0.5;
     public double accelerationPower = 0.1;
@@ -78,7 +_,7 @@
             }
 
             HitResult hitresult = ProjectileUtil.getHitResultOnMoveVector(this, this::canHitEntity, this.getClipType());
-            if (hitresult.getType() != HitResult.Type.MISS) {
+            if (hitresult.getType() != HitResult.Type.MISS && !net.neoforged.neoforge.event.EventHooks.onProjectileImpact(this, hitresult)) {
                 this.hitTargetOrDeflectSelf(hitresult);
             }
 
