--- a/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -77,6 +_,20 @@
     }
 
     protected AbstractArrow(
+            EntityType<? extends AbstractArrow> p_36717_,
+            double p_346045_,
+            double p_344855_,
+            double p_345999_,
+            Level p_36719_,
+            ItemStack p_309031_,
+            @Nullable ItemStack p_345487_
+            , @Nullable LivingEntity ownerEntity
+    ) {
+        this(p_36717_, p_346045_, p_344855_, p_345999_, p_36719_, p_309031_, p_345487_);
+        this.setOwner(ownerEntity);
+    }
+
+    protected AbstractArrow(
         EntityType<? extends AbstractArrow> p_36717_,
         double p_346045_,
         double p_344855_,
@@ -112,8 +_,8 @@
     protected AbstractArrow(
         EntityType<? extends AbstractArrow> p_36721_, LivingEntity p_345310_, Level p_36722_, ItemStack p_309145_, @Nullable ItemStack p_345000_
     ) {
-        this(p_36721_, p_345310_.getX(), p_345310_.getEyeY() - 0.1F, p_345310_.getZ(), p_36722_, p_309145_, p_345000_);
-        this.setOwner(p_345310_);
+        this(p_36721_, p_345310_.getX(), p_345310_.getEyeY() - 0.1F, p_345310_.getZ(), p_36722_, p_309145_, p_345000_, p_345310_); // CraftBukkit
+        // this.setOwner(p_345310_); // SPIGOT-7744 - Moved to the above constructor
     }
 
     public void setSoundEvent(SoundEvent p_36741_) {
@@ -188,7 +_,7 @@
             this.shakeTime--;
         }
 
-        if (this.isInWaterOrRain() || blockstate.is(Blocks.POWDER_SNOW)) {
+        if (this.isInWaterOrRain() || blockstate.is(Blocks.POWDER_SNOW) || this.isInFluidType((fluidType, height) -> this.canFluidExtinguish(fluidType))) {
             this.clearFire();
         }
 
@@ -224,7 +_,9 @@
                     }
                 }
 
-                if (hitresult != null && !flag) {
+                if (hitresult != null && hitresult.getType() != HitResult.Type.MISS && !flag) {
+                    if (net.neoforged.neoforge.event.EventHooks.onProjectileImpact(this, hitresult))
+                        break;
                     ProjectileDeflection projectiledeflection = this.hitTargetOrDeflectSelf(hitresult);
                     this.hasImpulse = true;
                     if (projectiledeflection != ProjectileDeflection.NONE) {
