--- a/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ b/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -35,19 +_,19 @@
 import org.jspecify.annotations.Nullable;
 
 public class FireworkRocketEntity extends Projectile implements ItemSupplier {
-    private static final EntityDataAccessor<ItemStack> DATA_ID_FIREWORKS_ITEM = SynchedEntityData.defineId(
+    public static final EntityDataAccessor<ItemStack> DATA_ID_FIREWORKS_ITEM = SynchedEntityData.defineId(
         FireworkRocketEntity.class, EntityDataSerializers.ITEM_STACK
     );
     private static final EntityDataAccessor<OptionalInt> DATA_ATTACHED_TO_TARGET = SynchedEntityData.defineId(
         FireworkRocketEntity.class, EntityDataSerializers.OPTIONAL_UNSIGNED_INT
     );
-    private static final EntityDataAccessor<Boolean> DATA_SHOT_AT_ANGLE = SynchedEntityData.defineId(FireworkRocketEntity.class, EntityDataSerializers.BOOLEAN);
+    public static final EntityDataAccessor<Boolean> DATA_SHOT_AT_ANGLE = SynchedEntityData.defineId(FireworkRocketEntity.class, EntityDataSerializers.BOOLEAN);
     private static final int DEFAULT_LIFE = 0;
     private static final int DEFAULT_LIFE_TIME = 0;
     private static final boolean DEFAULT_SHOT_AT_ANGLE = false;
-    private int life = 0;
-    private int lifetime = 0;
-    private @Nullable LivingEntity attachedToEntity;
+    public int life = 0;
+    public int lifetime = 0;
+    public @Nullable LivingEntity attachedToEntity;
 
     public FireworkRocketEntity(EntityType<? extends FireworkRocketEntity> type, Level level) {
         super(type, level);
@@ -183,6 +_,13 @@
 
         if (this.life > this.lifetime && this.level() instanceof ServerLevel level) {
             this.explode(level);
+        }
+    }
+
+    @Override
+    protected void onHit(HitResult result) {
+        if (result.getType() == HitResult.Type.MISS || !net.neoforged.neoforge.event.EventHooks.onProjectileImpact(this, result)) {
+            super.onHit(result);
         }
     }
 
