--- a/net/minecraft/world/entity/projectile/FishingHook.java
+++ b/net/minecraft/world/entity/projectile/FishingHook.java
@@ -55,16 +_,16 @@
     private boolean biting;
     private int outOfWaterTime;
     private static final int MAX_OUT_OF_WATER_TIME = 10;
-    private static final EntityDataAccessor<Integer> DATA_HOOKED_ENTITY = SynchedEntityData.defineId(FishingHook.class, EntityDataSerializers.INT);
+    public static final EntityDataAccessor<Integer> DATA_HOOKED_ENTITY = SynchedEntityData.defineId(FishingHook.class, EntityDataSerializers.INT);
     private static final EntityDataAccessor<Boolean> DATA_BITING = SynchedEntityData.defineId(FishingHook.class, EntityDataSerializers.BOOLEAN);
     private int life;
     private int nibble;
-    private int timeUntilLured;
-    private int timeUntilHooked;
+    public int timeUntilLured;
+    public int timeUntilHooked;
     private float fishAngle;
     private boolean openWater = true;
-    private @Nullable Entity hookedIn;
-    private FishingHook.FishHookState currentState = FishingHook.FishHookState.FLYING;
+    public @Nullable Entity hookedIn;
+    public FishingHook.FishHookState currentState = FishingHook.FishHookState.FLYING;
     private final int luck;
     private final int lureSpeed;
     private final InterpolationHandler interpolationHandler = new InterpolationHandler(this);
@@ -254,8 +_,8 @@
         if (owner.canInteractWithLevel()) {
             ItemStack selectedItem = owner.getMainHandItem();
             ItemStack selectedItemOffHand = owner.getOffhandItem();
-            boolean mainHandIsFishing = selectedItem.is(Items.FISHING_ROD);
-            boolean offHandIsFishing = selectedItemOffHand.is(Items.FISHING_ROD);
+            boolean mainHandIsFishing = selectedItem.canPerformAction(net.neoforged.neoforge.common.ItemAbilities.FISHING_ROD_CAST);
+            boolean offHandIsFishing = selectedItemOffHand.canPerformAction(net.neoforged.neoforge.common.ItemAbilities.FISHING_ROD_CAST);
             if ((mainHandIsFishing || offHandIsFishing) && this.distanceToSqr(owner) <= 1024.0) {
                 return false;
             }
@@ -267,7 +_,7 @@
 
     private void checkCollision() {
         HitResult hitResult = ProjectileUtil.getHitResultOnMoveVector(this, this::canHitEntity);
-        this.hitTargetOrDeflectSelf(hitResult);
+        if (hitResult.getType() == HitResult.Type.MISS || !net.neoforged.neoforge.event.EventHooks.onProjectileImpact(this, hitResult)) this.onHit(hitResult);
     }
 
     @Override
@@ -289,7 +_,7 @@
         this.setDeltaMovement(this.getDeltaMovement().normalize().scale(hitResult.distanceTo(this)));
     }
 
-    private void setHookedEntity(@Nullable Entity hookedIn) {
+    public void setHookedEntity(@Nullable Entity hookedIn) {
         this.hookedIn = hookedIn;
         this.getEntityData().set(DATA_HOOKED_ENTITY, hookedIn == null ? 0 : hookedIn.getId() + 1);
     }
@@ -448,6 +_,7 @@
         Player owner = this.getPlayerOwner();
         if (!this.level().isClientSide() && owner != null && !this.shouldStopFishing(owner)) {
             int dmg = 0;
+            net.neoforged.neoforge.event.entity.player.ItemFishedEvent event = null;
             if (this.hookedIn != null) {
                 this.pullEntity(this.hookedIn);
                 CriteriaTriggers.FISHING_ROD_HOOKED.trigger((ServerPlayer)owner, rod, this, Collections.emptyList());
@@ -458,10 +_,17 @@
                     .withParameter(LootContextParams.ORIGIN, this.position())
                     .withParameter(LootContextParams.TOOL, rod)
                     .withParameter(LootContextParams.THIS_ENTITY, this)
+                    .withParameter(LootContextParams.ATTACKING_ENTITY, this.getOwner())
                     .withLuck(this.luck + owner.getLuck())
                     .create(LootContextParamSets.FISHING);
                 LootTable lootTable = this.level().getServer().reloadableRegistries().getLootTable(BuiltInLootTables.FISHING);
                 List<ItemStack> items = lootTable.getRandomItems(params);
+                event = new net.neoforged.neoforge.event.entity.player.ItemFishedEvent(items, this.onGround() ? 2 : 1, this);
+                net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(event);
+                if (event.isCanceled()) {
+                    this.discard();
+                    return event.getRodDamage();
+                }
                 CriteriaTriggers.FISHING_ROD_HOOKED.trigger((ServerPlayer)owner, rod, this, items);
 
                 for (ItemStack itemStack : items) {
@@ -487,6 +_,7 @@
             }
 
             this.discard();
+            if (event != null) return event.getRodDamage();
             return dmg;
         } else {
             return 0;
@@ -502,7 +_,7 @@
         super.handleEntityEvent(id);
     }
 
-    protected void pullEntity(Entity entity) {
+    public void pullEntity(Entity entity) {
         Entity owner = this.getOwner();
         if (owner != null) {
             Vec3 delta = new Vec3(owner.getX() - this.getX(), owner.getY() - this.getY(), owner.getZ() - this.getZ()).scale(0.1);
@@ -568,7 +_,7 @@
         }
     }
 
-    private enum FishHookState {
+    public enum FishHookState {
         FLYING,
         HOOKED_IN_ENTITY,
         BOBBING;
