--- a/net/minecraft/world/inventory/ContainerLevelAccess.java
+++ b/net/minecraft/world/inventory/ContainerLevelAccess.java
@@ -5,8 +_,9 @@
 import java.util.function.BiFunction;
 import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.Level;
+import org.taiyitistmc.injection.world.inventory.InjectionContainerLevelAccess;
 
-public interface ContainerLevelAccess {
+public interface ContainerLevelAccess extends InjectionContainerLevelAccess {
     ContainerLevelAccess NULL = new ContainerLevelAccess() {
         @Override
         public <T> Optional<T> evaluate(BiFunction<Level, BlockPos, T> p_39304_) {
