--- a/net/minecraft/world/inventory/DispenserMenu.java
+++ b/net/minecraft/world/inventory/DispenserMenu.java
@@ -5,6 +_,7 @@
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
 
 public class DispenserMenu extends AbstractContainerMenu {
     private static final int SLOT_COUNT = 9;
@@ -13,6 +_,8 @@
     private static final int USE_ROW_SLOT_START = 36;
     private static final int USE_ROW_SLOT_END = 45;
     public final Container dispenser;
+    private CraftInventoryView bukkitEntity = null;
+    private Inventory player;
 
     public DispenserMenu(int p_39433_, Inventory p_39434_) {
         this(p_39433_, p_39434_, new SimpleContainer(9));
