--- a/net/minecraft/world/inventory/FurnaceMenu.java
+++ b/net/minecraft/world/inventory/FurnaceMenu.java
@@ -5,6 +_,7 @@
 import net.minecraft.world.item.crafting.RecipeType;
 
 public class FurnaceMenu extends AbstractFurnaceMenu {
+
     public FurnaceMenu(int p_39532_, Inventory p_39533_) {
         super(MenuType.FURNACE, RecipeType.SMELTING, RecipeBookType.FURNACE, p_39532_, p_39533_);
     }
