--- a/net/minecraft/world/inventory/ItemCombinerMenu.java
+++ b/net/minecraft/world/inventory/ItemCombinerMenu.java
@@ -45,11 +_,11 @@
     }
 
     private void createInputSlots(ItemCombinerMenuSlotDefinition itemInputSlots) {
-        for (final ItemCombinerMenuSlotDefinition.SlotDefinition slot : itemInputSlots.getSlots()) {
-            this.addSlot(new Slot(this.inputSlots, slot.slotIndex(), slot.x(), slot.y()) {
+        for (final ItemCombinerMenuSlotDefinition.SlotDefinition slotDefinition : itemInputSlots.getSlots()) {
+            this.addSlot(new Slot(this.inputSlots, slotDefinition.slotIndex(), slotDefinition.x(), slotDefinition.y()) {
                 @Override
                 public boolean mayPlace(ItemStack itemStack) {
-                    return slot.mayPlace().test(itemStack);
+                    return slotDefinition.mayPlace().test(itemStack);
                 }
             });
         }
