--- a/net/minecraft/world/inventory/LecternMenu.java
+++ b/net/minecraft/world/inventory/LecternMenu.java
@@ -2,8 +_,13 @@
 
 import net.minecraft.world.Container;
 import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.craftbukkit.inventory.CraftInventoryLectern;
+import org.bukkit.craftbukkit.inventory.view.CraftLecternView;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.inventory.PlayerInventory;
 
 public class LecternMenu extends AbstractContainerMenu {
     private static final int DATA_COUNT = 1;
@@ -14,9 +_,32 @@
     public static final int BUTTON_PAGE_JUMP_RANGE_START = 100;
     private final Container lectern;
     private final ContainerData lecternData;
-
+    // CraftBukkit start
+    private CraftLecternView bukkitEntity = null;
+    private HumanEntity player;
+
+    @Override
+    public CraftLecternView getBukkitView() {
+        if (bukkitEntity != null) {
+            return bukkitEntity;
+        }
+
+        CraftInventoryLectern inventory = new CraftInventoryLectern(this.lectern);
+        bukkitEntity = new CraftLecternView(this.player, inventory, this);
+        return bukkitEntity;
+    }
+    // CraftBukkit end
+
+    public LecternMenu(int p_39822_, Inventory playerinventory) {
+        this(p_39822_, new SimpleContainer(1), new SimpleContainerData(1), playerinventory);
+    }
     public LecternMenu(int p_39822_) {
         this(p_39822_, new SimpleContainer(1), new SimpleContainerData(1));
+    }
+
+    public LecternMenu(int p_39824_, Container p_39825_, ContainerData p_39826_, Inventory playerinventory) {
+        this(p_39824_, p_39825_, p_39826_);
+        playerinventory.player.getBukkitEntity(); // CraftBukkit
     }
 
     public LecternMenu(int p_39824_, Container p_39825_, ContainerData p_39826_) {
