--- a/net/minecraft/world/inventory/LoomMenu.java
+++ b/net/minecraft/world/inventory/LoomMenu.java
@@ -23,6 +_,7 @@
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.entity.BannerPattern;
 import net.minecraft.world.level.block.entity.BannerPatternLayers;
+import org.bukkit.craftbukkit.inventory.view.CraftLoomView;
 
 public class LoomMenu extends AbstractContainerMenu {
     private static final int PATTERN_NOT_SET = -1;
@@ -56,6 +_,8 @@
             LoomMenu.this.slotUpdateListener.run();
         }
     };
+    private CraftLoomView bukkitEntity;
+    private Inventory playerInventory;
 
     public LoomMenu(int p_39856_, Inventory p_39857_) {
         this(p_39856_, p_39857_, ContainerLevelAccess.NULL);
