--- a/net/minecraft/world/entity/vehicle/minecart/AbstractMinecartContainer.java
+++ b/net/minecraft/world/entity/vehicle/minecart/AbstractMinecartContainer.java
@@ -24,8 +_,8 @@
 
 public abstract class AbstractMinecartContainer extends AbstractMinecart implements ContainerEntity {
     private NonNullList<ItemStack> itemStacks = NonNullList.withSize(36, ItemStack.EMPTY);
-    public @Nullable ResourceKey<LootTable> lootTable;
-    public long lootTableSeed;
+    private @Nullable ResourceKey<LootTable> lootTable;
+    private long lootTableSeed;
 
     protected AbstractMinecartContainer(EntityType<?> type, Level level) {
         super(type, level);
@@ -94,6 +_,8 @@
 
     @Override
     public InteractionResult interact(Player player, InteractionHand hand, Vec3 location) {
+        InteractionResult ret = super.interact(player, hand, location);
+        if (ret.consumesAction()) return ret;
         return this.interactWithContainerVehicle(player);
     }
 
