--- a/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java
+++ b/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java
@@ -25,7 +_,7 @@
 import net.minecraft.world.phys.Vec3;
 
 public class MinecartCommandBlock extends AbstractMinecart {
-    private static final EntityDataAccessor<String> DATA_ID_COMMAND_NAME = SynchedEntityData.defineId(MinecartCommandBlock.class, EntityDataSerializers.STRING);
+    public static final EntityDataAccessor<String> DATA_ID_COMMAND_NAME = SynchedEntityData.defineId(MinecartCommandBlock.class, EntityDataSerializers.STRING);
     private static final EntityDataAccessor<Component> DATA_ID_LAST_OUTPUT = SynchedEntityData.defineId(
         MinecartCommandBlock.class, EntityDataSerializers.COMPONENT
     );
@@ -87,6 +_,8 @@
 
     @Override
     public InteractionResult interact(Player player, InteractionHand hand, Vec3 location) {
+        InteractionResult ret = super.interact(player, hand, location);
+        if (ret.consumesAction()) return ret;
         if (!player.canUseGameMasterBlocks()) {
             return InteractionResult.PASS;
         }
