--- a/net/minecraft/world/entity/vehicle/minecart/MinecartFurnace.java
+++ b/net/minecraft/world/entity/vehicle/minecart/MinecartFurnace.java
@@ -28,7 +_,7 @@
     private static final int MAX_FUEL_TICKS = 32000;
     private static final short DEFAULT_FUEL = 0;
     private static final Vec3 DEFAULT_PUSH = Vec3.ZERO;
-    private int fuel = 0;
+    public int fuel = 0;
     public Vec3 push = DEFAULT_PUSH;
 
     public MinecartFurnace(EntityType<? extends MinecartFurnace> type, Level level) {
@@ -107,6 +_,8 @@
 
     @Override
     public InteractionResult interact(Player player, InteractionHand hand, Vec3 location) {
+        InteractionResult ret = super.interact(player, hand, location);
+        if (ret.consumesAction()) return ret;
         ItemStack itemStack = player.getItemInHand(hand);
         if (this.addFuel(player.position(), itemStack)) {
             itemStack.consume(1, player);
