--- a/net/minecraft/network/protocol/common/custom/DiscardedPayload.java
+++ b/net/minecraft/network/protocol/common/custom/DiscardedPayload.java
@@ -3,8 +_,9 @@
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.Identifier;
+import org.teneted.neotenet.injection.network.network.protocol.common.custom.DiscardedPayloadInjection;
 
-public record DiscardedPayload(Identifier id) implements CustomPacketPayload {
+public record DiscardedPayload(Identifier id) implements CustomPacketPayload, DiscardedPayloadInjection {
     public static <T extends FriendlyByteBuf> StreamCodec<T, DiscardedPayload> codec(Identifier id, int maxPayloadSize) {
         return CustomPacketPayload.codec((payload, buf) -> {}, buf -> {
             int length = buf.readableBytes();
