--- a/net/minecraft/network/protocol/common/custom/DiscardedPayload.java
+++ b/net/minecraft/network/protocol/common/custom/DiscardedPayload.java
@@ -1,10 +_,24 @@
 package net.minecraft.network.protocol.common.custom;
 
+import io.netty.buffer.ByteBuf;
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.ResourceLocation;
 
-public record DiscardedPayload(ResourceLocation id) implements CustomPacketPayload {
+public final class DiscardedPayload implements CustomPacketPayload {
+
+    private ByteBuf data;
+    private final ResourceLocation id;
+
+    public DiscardedPayload(ResourceLocation p_320106_, ByteBuf data) {
+        this(p_320106_);
+        this.data = data;
+    }
+
+    public DiscardedPayload(ResourceLocation p_320106_) {
+        this.id = p_320106_;
+    }
+
     public static <T extends FriendlyByteBuf> StreamCodec<T, DiscardedPayload> codec(ResourceLocation p_320106_, int p_319929_) {
         return CustomPacketPayload.codec((p_320462_, p_319882_) -> {
         }, p_319935_ -> {
