--- a/net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
+++ b/net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
@@ -19,11 +_,12 @@
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.command.CommandSender;
 
 public class MinecartCommandBlock extends AbstractMinecart {
-    public static final EntityDataAccessor<String> DATA_ID_COMMAND_NAME = SynchedEntityData.defineId(MinecartCommandBlock.class, EntityDataSerializers.STRING);
+    static final EntityDataAccessor<String> DATA_ID_COMMAND_NAME = SynchedEntityData.defineId(MinecartCommandBlock.class, EntityDataSerializers.STRING);
     static final EntityDataAccessor<Component> DATA_ID_LAST_OUTPUT = SynchedEntityData.defineId(MinecartCommandBlock.class, EntityDataSerializers.COMPONENT);
-    private final BaseCommandBlock commandBlock = new MinecartCommandBlock.MinecartCommandBase();
+    private final BaseCommandBlock commandBlock = new MinecartCommandBase();
     private static final int ACTIVATION_DELAY = 4;
     private int lastActivated;
 
@@ -62,8 +_,8 @@
     }
 
     @Override
-    public AbstractMinecart.Type getMinecartType() {
-        return AbstractMinecart.Type.COMMAND_BLOCK;
+    public Type getMinecartType() {
+        return Type.COMMAND_BLOCK;
     }
 
     @Override
@@ -145,6 +_,11 @@
         @Override
         public boolean isValid() {
             return !MinecartCommandBlock.this.isRemoved();
+        }
+
+        @Override
+        public CommandSender getBukkitSender(CommandSourceStack wrapper) {
+            return (org.bukkit.craftbukkit.entity.CraftMinecartCommand) MinecartCommandBlock.this.getBukkitEntity();
         }
     }
 }
