--- a/net/minecraft/world/entity/monster/piglin/AbstractPiglin.java
+++ b/net/minecraft/world/entity/monster/piglin/AbstractPiglin.java
@@ -30,7 +_,7 @@
     private static final boolean DEFAULT_IMMUNE_TO_ZOMBIFICATION = false;
     private static final boolean DEFAULT_PICK_UP_LOOT = true;
     private static final int DEFAULT_TIME_IN_OVERWORLD = 0;
-    protected int timeInOverworld = 0;
+    public int timeInOverworld = 0;
 
     public AbstractPiglin(EntityType<? extends AbstractPiglin> type, Level level) {
         super(type, level);
@@ -52,7 +_,7 @@
         this.getEntityData().set(DATA_IMMUNE_TO_ZOMBIFICATION, isImmuneToZombification);
     }
 
-    protected boolean isImmuneToZombification() {
+    public boolean isImmuneToZombification() {
         return this.getEntityData().get(DATA_IMMUNE_TO_ZOMBIFICATION);
     }
 
@@ -86,7 +_,7 @@
             this.timeInOverworld = 0;
         }
 
-        if (this.timeInOverworld > 300) {
+        if (this.timeInOverworld > 300 && net.neoforged.neoforge.event.EventHooks.canLivingConvert(this, EntityTypes.ZOMBIFIED_PIGLIN, (timer) -> this.timeInOverworld = timer)) {
             if (level.getDifficulty() != Difficulty.PEACEFUL) {
                 this.playConvertedSound();
             }
@@ -110,7 +_,10 @@
         this.convertTo(
             EntityTypes.ZOMBIFIED_PIGLIN,
             ConversionParams.single(this, true, true),
-            zombified -> zombified.addEffect(new MobEffectInstance(MobEffects.NAUSEA, 200, 0))
+            zombified -> {
+                zombified.addEffect(new MobEffectInstance(MobEffects.NAUSEA, 200, 0));
+                net.neoforged.neoforge.event.EventHooks.onLivingConvert(this, zombified);
+            }
         );
     }
 
