--- a/net/minecraft/world/entity/monster/illager/Vindicator.java
+++ b/net/minecraft/world/entity/monster/illager/Vindicator.java
@@ -51,7 +_,7 @@
     private static final String TAG_JOHNNY = "Johnny";
     private static final Predicate<Difficulty> DOOR_BREAKING_PREDICATE = d -> d == Difficulty.NORMAL || d == Difficulty.HARD;
     private static final boolean DEFAULT_JOHNNY = false;
-    public boolean isJohnny = false;
+    private boolean isJohnny = false;
 
     public Vindicator(EntityType<? extends Vindicator> type, Level level) {
         super(type, level);
