--- a/net/minecraft/world/level/portal/PortalForcer.java
+++ b/net/minecraft/world/level/portal/PortalForcer.java
@@ -5,7 +_,6 @@
 import net.minecraft.BlockUtil;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
-import net.minecraft.core.Holder;
 import net.minecraft.core.Vec3i;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.util.Mth;
@@ -18,8 +_,9 @@
 import net.minecraft.world.level.block.state.properties.BlockStateProperties;
 import net.minecraft.world.level.border.WorldBorder;
 import net.minecraft.world.level.levelgen.Heightmap;
+import org.teneted.neotenet.injection.world.level.portal.InjectionPortalForcer;
 
-public class PortalForcer {
+public class PortalForcer implements InjectionPortalForcer {
     public static final int TICKET_RADIUS = 3;
     private static final int NETHER_PORTAL_RADIUS = 16;
     private static final int OVERWORLD_PORTAL_RADIUS = 128;
