--- a/net/minecraft/core/component/DataComponentPatch.java
+++ b/net/minecraft/core/component/DataComponentPatch.java
@@ -17,6 +_,7 @@
 import net.minecraft.resources.Identifier;
 import net.minecraft.util.Unit;
 import org.jspecify.annotations.Nullable;
+import org.teneted.neotenet.injection.core.component.DataComponentPatch_BuilderInjection;
 
 public final class DataComponentPatch {
     public static final DataComponentPatch EMPTY = new DataComponentPatch(Reference2ObjectMaps.emptyMap());
@@ -156,6 +_,14 @@
         return new DataComponentPatch.Builder();
     }
 
+    /**
+     * {@return the patch stored for the given component type, null if no patch is stored}
+     */
+    @SuppressWarnings("unchecked")
+    public <T> @Nullable Optional<T> getPatch(DataComponentType<? extends T> type) {
+        return (Optional<T>) this.map.get(type);
+    }
+
     public <T> @Nullable T get(DataComponentGetter prototype, DataComponentType<? extends T> type) {
         return getFromPatchAndPrototype(this.map, prototype, type);
     }
@@ -248,13 +_,14 @@
         return builder.toString();
     }
 
-    public static class Builder {
+    public static class Builder implements DataComponentPatch_BuilderInjection {
         private final Reference2ObjectMap<DataComponentType<?>, Optional<?>> map = new Reference2ObjectArrayMap<>();
 
         private Builder() {
         }
 
         public <T> DataComponentPatch.Builder set(DataComponentType<T> type, T value) {
+            net.neoforged.neoforge.common.CommonHooks.validateComponent(value);
             this.map.put(type, Optional.of(value));
             return this;
         }
