--- a/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java
+++ b/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java
@@ -16,7 +_,7 @@
 
 public class SpectralArrow extends AbstractArrow {
     private static final int DEFAULT_DURATION = 200;
-    private int duration = 200;
+    public int duration = 200;
 
     public SpectralArrow(EntityType<? extends SpectralArrow> type, Level level) {
         super(type, level);
