--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -50,8 +_,9 @@
 import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import org.taiyitistmc.injection.world.level.block.InjectionChestBlock;
 
-public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements SimpleWaterloggedBlock {
+public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements SimpleWaterloggedBlock, InjectionChestBlock {
     public static final MapCodec<ChestBlock> CODEC = simpleCodec(p_304364_ -> new ChestBlock(p_304364_, () -> BlockEntityType.CHEST));
     public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;
     public static final EnumProperty<ChestType> TYPE = BlockStateProperties.CHEST_TYPE;
@@ -357,7 +_,8 @@
 
     @Override
     protected BlockState mirror(BlockState p_51549_, Mirror p_51550_) {
-        return p_51549_.rotate(p_51550_.getRotation(p_51549_.getValue(FACING)));
+        BlockState rotated = p_51549_.rotate(p_51550_.getRotation(p_51549_.getValue(FACING)));
+        return p_51550_ == Mirror.NONE ? rotated : rotated.setValue(TYPE, rotated.getValue(TYPE).getOpposite());  // Forge: Fixed MC-134110 Structure mirroring breaking apart double chests
     }
 
     @Override
