--- a/net/minecraft/world/level/block/CopperBulbBlock.java
+++ b/net/minecraft/world/level/block/CopperBulbBlock.java
@@ -28,7 +_,7 @@
     }
 
     @Override
-    protected void onPlace(BlockState p_309207_, Level p_309068_, BlockPos p_309087_, BlockState p_308908_, boolean p_308906_) {
+    public void onPlace(BlockState p_309207_, Level p_309068_, BlockPos p_309087_, BlockState p_308908_, boolean p_308906_) {
         if (p_308908_.getBlock() != p_309207_.getBlock() && p_309068_ instanceof ServerLevel serverlevel) {
             this.checkAndFlip(p_309207_, serverlevel, p_309087_);
         }
