--- a/net/minecraft/world/level/block/FallingBlock.java
+++ b/net/minecraft/world/level/block/FallingBlock.java
@@ -25,7 +_,7 @@
     protected abstract MapCodec<? extends FallingBlock> codec();
 
     @Override
-    protected void onPlace(BlockState p_53233_, Level p_53234_, BlockPos p_53235_, BlockState p_53236_, boolean p_53237_) {
+    public void onPlace(BlockState p_53233_, Level p_53234_, BlockPos p_53235_, BlockState p_53236_, boolean p_53237_) {
         p_53234_.scheduleTick(p_53235_, this, this.getDelayAfterPlace());
     }
 
