--- a/net/minecraft/world/level/block/SoulSandBlock.java
+++ b/net/minecraft/world/level/block/SoulSandBlock.java
@@ -59,7 +_,7 @@
     }
 
     @Override
-    protected void onPlace(BlockState p_56696_, Level p_56697_, BlockPos p_56698_, BlockState p_56699_, boolean p_56700_) {
+    public void onPlace(BlockState p_56696_, Level p_56697_, BlockPos p_56698_, BlockState p_56699_, boolean p_56700_) {
         p_56697_.scheduleTick(p_56698_, this, 20);
     }
 
