--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -29,7 +_,7 @@
     }
 
     @Override
-    protected void onPlace(BlockState p_56811_, Level p_56812_, BlockPos p_56813_, BlockState p_56814_, boolean p_56815_) {
+    public void onPlace(BlockState p_56811_, Level p_56812_, BlockPos p_56813_, BlockState p_56814_, boolean p_56815_) {
         if (!p_56814_.is(p_56811_.getBlock())) {
             this.tryAbsorbWater(p_56812_, p_56813_);
         }
@@ -49,6 +_,7 @@
     }
 
     private boolean removeWaterBreadthFirstSearch(Level p_56808_, BlockPos p_56809_) {
+        BlockState spongeState = p_56808_.getBlockState(p_56809_);
         return BlockPos.breadthFirstTraversal(
                 p_56809_,
                 6,
@@ -64,7 +_,7 @@
                     } else {
                         BlockState blockstate = p_56808_.getBlockState(p_294069_);
                         FluidState fluidstate = p_56808_.getFluidState(p_294069_);
-                        if (!fluidstate.is(FluidTags.WATER)) {
+                        if (!spongeState.canBeHydrated(p_56808_, p_56809_, fluidstate, p_294069_)) {
                             return false;
                         } else {
                             if (blockstate.getBlock() instanceof BucketPickup bucketpickup
