--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -52,7 +_,7 @@
     private static final int MAX_TICKS_BEFORE_LOGIN = 600;
     private final byte[] challenge;
     private final MinecraftServer server;
-    private final Connection connection;
+    public final Connection connection;
     private final ServerActivityMonitor serverActivityMonitor;
     private volatile ServerLoginPacketListenerImpl.State state = ServerLoginPacketListenerImpl.State.HELLO;
     private int tick;
