--- a/net/minecraft/world/damagesource/DamageSource.java
+++ b/net/minecraft/world/damagesource/DamageSource.java
@@ -11,8 +_,9 @@
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.phys.Vec3;
+import org.celestial_artistry.neotenet.injection.world.damagesource.InjectionDamageSource;
 
-public class DamageSource {
+public class DamageSource implements InjectionDamageSource {
     private final Holder<DamageType> type;
     @Nullable
     private final Entity causingEntity;
@@ -93,6 +_,10 @@
         return this.type().msgId();
     }
 
+    /**
+     * @deprecated Use {@link DamageScaling#getScalingFunction()}
+     */
+    @Deprecated(since = "1.20.1")
     public boolean scalesWithDifficulty() {
         return switch (this.type().scaling()) {
             case NEVER -> false;
