--- a/net/minecraft/world/damagesource/DamageSources.java
+++ b/net/minecraft/world/damagesource/DamageSources.java
@@ -14,8 +_,9 @@
 import net.minecraft.world.entity.projectile.WitherSkull;
 import net.minecraft.world.level.Explosion;
 import net.minecraft.world.phys.Vec3;
+import org.celestial_artistry.neotenet.injection.world.damagesource.InjectionDamageSources;
 
-public class DamageSources {
+public class DamageSources implements InjectionDamageSources {
     public final Registry<DamageType> damageTypes;
     private final DamageSource inFire;
     private final DamageSource campfire;
@@ -41,6 +_,9 @@
     private final DamageSource stalagmite;
     private final DamageSource outsideBorder;
     private final DamageSource genericKill;
+    // CraftBukkit start
+    private DamageSource melting;
+    private DamageSource poison;
 
     public DamageSources(RegistryAccess p_270740_) {
         this.damageTypes = p_270740_.registryOrThrow(Registries.DAMAGE_TYPE);
