--- a/net/minecraft/world/item/crafting/RecipeHolder.java
+++ b/net/minecraft/world/item/crafting/RecipeHolder.java
@@ -3,8 +_,9 @@
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.ResourceLocation;
+import org.teneted.neotenet.injection.world.item.crafting.InjectionRecipeHolder;
 
-public record RecipeHolder<T extends Recipe<?>>(ResourceLocation id, T value) {
+public record RecipeHolder<T extends Recipe<?>>(ResourceLocation id, T value) implements InjectionRecipeHolder {
     public static final StreamCodec<RegistryFriendlyByteBuf, RecipeHolder<?>> STREAM_CODEC = StreamCodec.composite(
         ResourceLocation.STREAM_CODEC, RecipeHolder::id, Recipe.STREAM_CODEC, RecipeHolder::value, RecipeHolder::new
     );
