--- a/net/minecraft/world/food/FoodData.java
+++ b/net/minecraft/world/food/FoodData.java
@@ -11,9 +_,9 @@
 public class FoodData {
     private static final int DEFAULT_TICK_TIMER = 0;
     private static final float DEFAULT_EXHAUSTION_LEVEL = 0.0F;
-    private int foodLevel = 20;
-    private float saturationLevel = 5.0F;
-    private float exhaustionLevel;
+    public int foodLevel = 20;
+    public float saturationLevel = 5.0F;
+    public float exhaustionLevel;
     private int tickTimer;
 
     private void add(int food, float saturation) {
