--- a/net/minecraft/network/chat/Component.java
+++ b/net/minecraft/network/chat/Component.java
@@ -19,9 +_,11 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Date;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Stream;
 import javax.annotation.Nullable;
 import net.minecraft.core.HolderLookup;
 import net.minecraft.network.chat.contents.DataSource;
@@ -34,8 +_,22 @@
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.FormattedCharSequence;
 import net.minecraft.world.level.ChunkPos;
-
-public interface Component extends Message, FormattedText {
+import org.taiyitistmc.injection.network.chat.InjectionComponent;
+
+public interface Component extends Message, FormattedText, InjectionComponent, Iterable<Component> { // CraftBukkit
+
+    // CraftBukkit start
+    @Override
+    default Stream<Component> stream() {
+        return com.google.common.collect.Streams.concat(new Stream[]{Stream.of(this), this.getSiblings().stream().flatMap(Component::stream)});
+    }
+
+    @Override
+    default Iterator<Component> iterator() {
+        return this.stream().iterator();
+    }
+    // CraftBukkit end
+
     Style getStyle();
 
     ComponentContents getContents();
