--- a/net/minecraft/network/chat/Component.java
+++ b/net/minecraft/network/chat/Component.java
@@ -26,8 +_,9 @@
 import net.minecraft.util.FormattedCharSequence;
 import net.minecraft.world.level.ChunkPos;
 import org.jspecify.annotations.Nullable;
+import org.teneted.neotenet.injection.network.chat.ComponentInjection;
 
-public interface Component extends Message, FormattedText {
+public interface Component extends Message, FormattedText, ComponentInjection {
     Style getStyle();
 
     ComponentContents getContents();
