--- a/net/minecraft/network/chat/TextColor.java
+++ b/net/minecraft/network/chat/TextColor.java
@@ -9,8 +_,9 @@
 import java.util.Objects;
 import net.minecraft.ChatFormatting;
 import org.jspecify.annotations.Nullable;
+import org.teneted.neotenet.injection.network.chat.TextColorInjection;
 
-public final class TextColor {
+public final class TextColor implements TextColorInjection {
     private static final String CUSTOM_COLOR_PREFIX = "#";
     public static final Codec<TextColor> CODEC = Codec.STRING.comapFlatMap(TextColor::parseColor, TextColor::serialize);
     private static final Map<String, TextColor> NAMED_COLORS = new HashMap<>();
@@ -31,16 +_,28 @@
     public static final TextColor YELLOW = named("yellow", 16777045);
     public static final TextColor WHITE = named("white", 16777215);
     private final int value;
-    public final @Nullable String name;
+    private final @Nullable String name;
+    // CraftBukkit start
+    @Nullable
+    public final ChatFormatting format;
+
+    private TextColor(int value, String name, ChatFormatting format) {
+        this.format = format;
+        // CraftBukkit end
+        this.value = value & 16777215;
+        this.name = name;
+    }
 
     private TextColor(int value, String name) {
         this.value = value & 16777215;
         this.name = name;
+        this.format = null;
     }
 
     private TextColor(int value) {
         this.value = value & 16777215;
         this.name = null;
+        this.format = null;
     }
 
     private static TextColor named(String name, int rgb) {
