--- a/net/minecraft/network/chat/TextColor.java
+++ b/net/minecraft/network/chat/TextColor.java
@@ -31,7 +_,7 @@
     public static final TextColor YELLOW = named("yellow", 16777045);
     public static final TextColor WHITE = named("white", 16777215);
     private final int value;
-    public final @Nullable String name;
+    private final @Nullable String name;
 
     private TextColor(int value, String name) {
         this.value = value & 16777215;
