--- a/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
@@ -33,8 +_,9 @@
 import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.Vec3;
 import org.jspecify.annotations.Nullable;
+import org.teneted.neotenet.injection.commands.arguments.selector.EntitySelectorParserInjection;
 
-public class EntitySelectorParser {
+public class EntitySelectorParser implements EntitySelectorParserInjection {
     public static final char SYNTAX_SELECTOR_START = '@';
     private static final char SYNTAX_OPTIONS_START = '[';
     private static final char SYNTAX_OPTIONS_END = ']';
@@ -465,6 +_,9 @@
             }
 
             this.reader.skip();
+            EntitySelector forgeSelector = net.neoforged.neoforge.common.command.EntitySelectorManager.parseSelector(this);
+            if (forgeSelector != null)
+                return forgeSelector;
             this.parseSelector();
         } else {
             this.parseNameOrUUID();
@@ -481,6 +_,7 @@
         builder.suggest("@s", Component.translatable("argument.entity.selector.self"));
         builder.suggest("@e", Component.translatable("argument.entity.selector.allEntities"));
         builder.suggest("@n", Component.translatable("argument.entity.selector.nearestEntity"));
+        net.neoforged.neoforge.common.command.EntitySelectorManager.fillSelectorSuggestions(builder);
     }
 
     private CompletableFuture<Suggestions> suggestNameOrSelector(SuggestionsBuilder builder, Consumer<SuggestionsBuilder> names) {
