--- a/net/minecraft/world/CompoundContainer.java
+++ b/net/minecraft/world/CompoundContainer.java
@@ -3,10 +_,14 @@
 import net.minecraft.world.entity.ContainerUser;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.entity.HumanEntity;
+
+import java.util.List;
 
 public class CompoundContainer implements Container {
     public final Container container1;
     public final Container container2;
+    public List<HumanEntity> transaction = new java.util.ArrayList<HumanEntity>();
 
     public CompoundContainer(Container container1, Container container2) {
         this.container1 = container1;
