--- a/net/minecraft/world/CompoundContainer.java
+++ b/net/minecraft/world/CompoundContainer.java
@@ -5,8 +_,8 @@
 import net.minecraft.world.item.ItemStack;
 
 public class CompoundContainer implements Container {
-    public final Container container1;
-    public final Container container2;
+    private final Container container1;
+    private final Container container2;
 
     public CompoundContainer(Container container1, Container container2) {
         this.container1 = container1;
