--- a/net/minecraft/world/SimpleContainer.java
+++ b/net/minecraft/world/SimpleContainer.java
@@ -12,12 +_,30 @@
 import net.minecraft.world.inventory.StackedContentsCompatible;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.entity.HumanEntity;
 
 public class SimpleContainer implements Container, StackedContentsCompatible {
     private final int size;
     public final NonNullList<ItemStack> items;
     @Nullable
     private List<ContainerListener> listeners;
+    // CraftBukkit start - add fields and methods
+    public List<HumanEntity> transaction = new java.util.ArrayList<HumanEntity>();
+    private int maxStack = MAX_STACK;
+    protected org.bukkit.inventory.InventoryHolder bukkitOwner;
+
+    public SimpleContainer(int i, org.bukkit.inventory.InventoryHolder owner) {
+        this(i);
+        this.bukkitOwner = owner;
+        // CraftBukkit end
+    }
+
+    public SimpleContainer(SimpleContainer original) {
+        this(original.size);
+        for (int slot = 0; slot < original.size; slot++) {
+            this.items.set(slot, original.items.get(slot).copy());
+        }
+    }
 
     public SimpleContainer(int p_19150_) {
         this.size = p_19150_;
