--- a/net/minecraft/world/entity/animal/nautilus/AbstractNautilus.java
+++ b/net/minecraft/world/entity/animal/nautilus/AbstractNautilus.java
@@ -78,7 +_,7 @@
     private static final float DASH_MOMENTUM_ON_LAND = 0.5F;
     private int dashCooldown = 0;
     protected float playerJumpPendingScale;
-    public SimpleContainer inventory;
+    protected SimpleContainer inventory;
     private static final double BUBBLE_SPREAD_FACTOR = 0.8;
     private static final double BUBBLE_DIRECTION_SCALE = 1.1;
     private static final double BUBBLE_Y_OFFSET = 0.25;
