--- a/net/minecraft/commands/CommandSource.java
+++ b/net/minecraft/commands/CommandSource.java
@@ -1,8 +_,9 @@
 package net.minecraft.commands;
 
 import net.minecraft.network.chat.Component;
+import org.taiyitistmc.injection.commands.InjectionCommandSource;
 
-public interface CommandSource {
+public interface CommandSource extends InjectionCommandSource {
     CommandSource NULL = new CommandSource() {
         @Override
         public void sendSystemMessage(Component p_230799_) {
@@ -22,6 +_,13 @@
         public boolean shouldInformAdmins() {
             return false;
         }
+
+        // CraftBukkit start
+        @Override
+        public org.bukkit.command.CommandSender getBukkitSender(CommandSourceStack wrapper) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // CraftBukkit end
     };
 
     void sendSystemMessage(Component p_230797_);
