--- a/net/minecraft/commands/CommandSourceStack.java
+++ b/net/minecraft/commands/CommandSourceStack.java
@@ -17,6 +_,8 @@
 import java.util.function.BinaryOperator;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
+
+import com.mojang.brigadier.tree.CommandNode;
 import net.minecraft.ChatFormatting;
 import net.minecraft.advancements.AdvancementHolder;
 import net.minecraft.commands.arguments.EntityAnchorArgument;
@@ -35,7 +_,9 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.server.permissions.Permission;
 import net.minecraft.server.permissions.PermissionSet;
+import net.minecraft.server.permissions.Permissions;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.util.Mth;
 import net.minecraft.util.TaskChainer;
@@ -49,8 +_,9 @@
 import org.jspecify.annotations.Nullable;
 
 import net.neoforged.neoforge.common.extensions.ICommandSourceStackExtension;
+import org.teneted.neotenet.injection.commands.CommandSourceStackInjection;
 
-public class CommandSourceStack implements SharedSuggestionProvider, ExecutionCommandSource<CommandSourceStack>, ICommandSourceStackExtension {
+public class CommandSourceStack implements SharedSuggestionProvider, ExecutionCommandSource<CommandSourceStack>, ICommandSourceStackExtension, CommandSourceStackInjection {
     public static final SimpleCommandExceptionType ERROR_NOT_PLAYER = new SimpleCommandExceptionType(Component.translatable("permissions.requires.player"));
     public static final SimpleCommandExceptionType ERROR_NOT_ENTITY = new SimpleCommandExceptionType(Component.translatable("permissions.requires.entity"));
     public final CommandSource source;
@@ -67,6 +_,7 @@
     private final Vec2 rotation;
     private final CommandSigningContext signingContext;
     private final TaskChainer chatMessageChainer;
+    public volatile CommandNode currentCommand;
 
     public CommandSourceStack(
         CommandSource source,
@@ -128,6 +_,37 @@
         this.signingContext = signingContext;
         this.chatMessageChainer = chatMessageChainer;
     }
+
+    public class DelegatePermissionSet implements PermissionSet {
+
+        private final PermissionSet handle;
+
+        public DelegatePermissionSet(PermissionSet handle) {
+            this.handle = handle;
+        }
+
+        @Override
+        public boolean hasPermission(Permission permission) {
+            boolean hasPermission = handle.hasPermission(permission);
+
+            CommandNode currentCommand = CommandSourceStack.this.currentCommand;
+            if (currentCommand != null) {
+                return hasPermission(hasPermission, org.bukkit.craftbukkit.command.VanillaCommandWrapper.getPermission(currentCommand));
+            }
+
+            if (permission.equals(Permissions.COMMANDS_ENTITY_SELECTORS)) {
+                return hasPermission(hasPermission, "minecraft.command.selector");
+            }
+
+            return hasPermission;
+        }
+
+        public boolean hasPermission(boolean hasPermission, String bukkitPermission) {
+            // World is null when loading functions
+            return ((getLevel() == null || !getLevel().getCraftServer().ignoreVanillaPermissions) && hasPermission) || getBukkitSender().hasPermission(bukkitPermission);
+        }
+    }
+    // CraftBukkit end
 
     public CommandSourceStack withSource(CommandSource source) {
         return this.source == source
