--- a/net/minecraft/world/entity/raid/Raids.java
+++ b/net/minecraft/world/entity/raid/Raids.java
@@ -41,7 +_,7 @@
             .apply(i, Raids::new)
     );
     public static final SavedDataType<Raids> TYPE = new SavedDataType<>(RAID_FILE_ID, Raids::new, CODEC, DataFixTypes.SAVED_DATA_RAIDS);
-    public final Int2ObjectMap<Raid> raidMap = new Int2ObjectOpenHashMap<>();
+    private final Int2ObjectMap<Raid> raidMap = new Int2ObjectOpenHashMap<>();
     private int nextId = 1;
     private int tick;
 
