--- a/net/minecraft/server/players/StoredUserList.java
+++ b/net/minecraft/server/players/StoredUserList.java
@@ -21,8 +_,9 @@
 import net.minecraft.Util;
 import net.minecraft.util.GsonHelper;
 import org.slf4j.Logger;
+import org.taiyitistmc.injection.server.players.InjectionStoredUserList;
 
-public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
+public abstract class StoredUserList<K, V extends StoredUserEntry<K>> implements InjectionStoredUserList<K, V> {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
     private final File file;
