--- a/net/minecraft/client/gui/screens/inventory/AbstractCommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/AbstractCommandBlockEditScreen.java
@@ -11,7 +_,7 @@
 import net.minecraft.network.chat.CommonComponents;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.MutableComponent;
-import net.minecraft.world.level.BaseCommandBlock;
+import net.minecraft.world.level.BaseCommandBlockCommandSource;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
@@ -38,7 +_,7 @@
         }
     }
 
-    abstract BaseCommandBlock getCommandBlock();
+    abstract BaseCommandBlockCommandSource getCommandBlock();
 
     abstract int getPreviousY();
 
@@ -56,7 +_,7 @@
                 .withInitialValue(flag)
                 .displayOnlyValue()
                 .create(this.width / 2 + 150 - 20, this.getPreviousY(), 20, 20, Component.translatable("advMode.trackOutput"), (p_169596_, p_169597_) -> {
-                    BaseCommandBlock basecommandblock = this.getCommandBlock();
+                    BaseCommandBlockCommandSource basecommandblock = this.getCommandBlock();
                     basecommandblock.setTrackOutput(p_169597_);
                     this.updatePreviousOutput(p_169597_);
                 })
@@ -104,7 +_,7 @@
     }
 
     protected void onDone() {
-        BaseCommandBlock basecommandblock = this.getCommandBlock();
+        BaseCommandBlockCommandSource basecommandblock = this.getCommandBlock();
         this.populateAndSendPacket(basecommandblock);
         if (!basecommandblock.isTrackOutput()) {
             basecommandblock.setLastOutput(null);
@@ -113,7 +_,7 @@
         this.minecraft.setScreen(null);
     }
 
-    protected abstract void populateAndSendPacket(BaseCommandBlock p_97670_);
+    protected abstract void populateAndSendPacket(BaseCommandBlockCommandSource p_97670_);
 
     private void onEdited(String p_97689_) {
         this.commandSuggestions.updateCommandInfo();
