--- a/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
@@ -5,7 +_,7 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket;
-import net.minecraft.world.level.BaseCommandBlock;
+import net.minecraft.world.level.BaseCommandBlockCommandSource;
 import net.minecraft.world.level.block.entity.CommandBlockEntity;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
@@ -25,7 +_,7 @@
     }
 
     @Override
-    BaseCommandBlock getCommandBlock() {
+    BaseCommandBlockCommandSource getCommandBlock() {
         return this.autoCommandBlock.getCommandBlock();
     }
 
@@ -76,7 +_,7 @@
     }
 
     public void updateGui() {
-        BaseCommandBlock basecommandblock = this.autoCommandBlock.getCommandBlock();
+        BaseCommandBlockCommandSource basecommandblock = this.autoCommandBlock.getCommandBlock();
         this.commandEdit.setValue(basecommandblock.getCommand());
         boolean flag = basecommandblock.isTrackOutput();
         this.mode = this.autoCommandBlock.getMode();
@@ -97,7 +_,7 @@
     }
 
     @Override
-    protected void populateAndSendPacket(BaseCommandBlock p_98384_) {
+    protected void populateAndSendPacket(BaseCommandBlockCommandSource p_98384_) {
         this.minecraft
             .getConnection()
             .send(
