--- a/net/minecraft/client/gui/screens/inventory/MinecartCommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/MinecartCommandBlockEditScreen.java
@@ -1,21 +_,20 @@
 package net.minecraft.client.gui.screens.inventory;
 
 import net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket;
-import net.minecraft.world.entity.vehicle.MinecartCommandBlock;
-import net.minecraft.world.level.BaseCommandBlock;
+import net.minecraft.world.level.BaseCommandBlockCommandSource;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
 public class MinecartCommandBlockEditScreen extends AbstractCommandBlockEditScreen {
-    private final BaseCommandBlock commandBlock;
+    private final BaseCommandBlockCommandSource commandBlock;
 
-    public MinecartCommandBlockEditScreen(BaseCommandBlock p_99216_) {
+    public MinecartCommandBlockEditScreen(BaseCommandBlockCommandSource p_99216_) {
         this.commandBlock = p_99216_;
     }
 
     @Override
-    public BaseCommandBlock getCommandBlock() {
+    public BaseCommandBlockCommandSource getCommandBlock() {
         return this.commandBlock;
     }
 
@@ -31,8 +_,8 @@
     }
 
     @Override
-    protected void populateAndSendPacket(BaseCommandBlock p_99218_) {
-        if (p_99218_ instanceof MinecartCommandBlock.MinecartCommandBase minecartcommandblock$minecartcommandbase) {
+    protected void populateAndSendPacket(BaseCommandBlockCommandSource p_99218_) {
+        if (p_99218_ instanceof MinecartCommandBlock.MinecartCommandBaseCommandSource minecartcommandblock$minecartcommandbase) {
             this.minecraft
                 .getConnection()
                 .send(
