--- a/net/minecraft/world/effect/MobEffectUtil.java
+++ b/net/minecraft/world/effect/MobEffectUtil.java
@@ -11,6 +_,7 @@
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.event.entity.EntityPotionEffectEvent;
 
 public final class MobEffectUtil {
     public static Component formatDuration(MobEffectInstance p_268116_, float p_268280_, float p_314720_) {
@@ -45,6 +_,25 @@
     }
 
     public static List<ServerPlayer> addEffectToPlayersAround(
+            ServerLevel p_216947_, @Nullable Entity p_216948_, Vec3 p_216949_, double p_216950_, MobEffectInstance p_216951_, int p_216952_
+    , org.bukkit.event.entity.EntityPotionEffectEvent.Cause cause ) {
+        Holder<MobEffect> holder = p_216951_.getEffect();
+        List<ServerPlayer> list = p_216947_.getPlayers(
+                p_267925_ -> p_267925_.gameMode.isSurvival()
+                        && (p_216948_ == null || !p_216948_.isAlliedTo(p_267925_))
+                        && p_216949_.closerThan(p_267925_.position(), p_216950_)
+                        && (
+                        !p_267925_.hasEffect(holder)
+                                || p_267925_.getEffect(holder).getAmplifier() < p_216951_.getAmplifier()
+                                || p_267925_.getEffect(holder).endsWithin(p_216952_ - 1)
+                )
+        );
+        list.forEach(p_238232_ -> p_238232_.pushEffectCause(cause));
+        list.forEach(p_238232_ -> p_238232_.addEffect(new MobEffectInstance(p_216951_), p_216948_));
+        return list;
+    }
+
+    public static List<ServerPlayer> addEffectToPlayersAround(
         ServerLevel p_216947_, @Nullable Entity p_216948_, Vec3 p_216949_, double p_216950_, MobEffectInstance p_216951_, int p_216952_
     ) {
         Holder<MobEffect> holder = p_216951_.getEffect();
@@ -58,6 +_,7 @@
                             || p_267925_.getEffect(holder).endsWithin(p_216952_ - 1)
                     )
         );
+        list.forEach(p_238232_ -> p_238232_.pushEffectCause(EntityPotionEffectEvent.Cause.UNKNOWN));
         list.forEach(p_238232_ -> p_238232_.addEffect(new MobEffectInstance(p_216951_), p_216948_));
         return list;
     }
