--- a/net/minecraft/world/entity/animal/feline/Ocelot.java
+++ b/net/minecraft/world/entity/animal/feline/Ocelot.java
@@ -72,11 +_,11 @@
         this.reassessTrustingGoals();
     }
 
-    private boolean isTrusting() {
+    public boolean isTrusting() {
         return this.entityData.get(DATA_TRUSTING);
     }
 
-    private void setTrusting(boolean trusting) {
+    public void setTrusting(boolean trusting) {
         this.entityData.set(DATA_TRUSTING, trusting);
         this.reassessTrustingGoals();
     }
@@ -168,7 +_,7 @@
         if ((this.temptGoal == null || this.temptGoal.isRunning()) && !this.isTrusting() && this.isFood(itemStack) && player.distanceToSqr(this) < 9.0) {
             this.usePlayerItem(player, hand, itemStack);
             if (!this.level().isClientSide()) {
-                if (this.random.nextInt(3) == 0) {
+                if (this.random.nextInt(3) == 0 && !net.neoforged.neoforge.event.EventHooks.onAnimalTame(this, player)) {
                     this.setTrusting(true);
                     this.spawnTrustingParticles(true);
                     this.level().broadcastEntityEvent(this, (byte)41);
