--- a/net/minecraft/world/entity/animal/axolotl/Axolotl.java
+++ b/net/minecraft/world/entity/animal/axolotl/Axolotl.java
@@ -281,7 +_,7 @@
         return Axolotl.Variant.byId(this.entityData.get(DATA_VARIANT));
     }
 
-    public void setVariant(Axolotl.Variant variant) {
+    private void setVariant(Axolotl.Variant variant) {
         this.entityData.set(DATA_VARIANT, variant.getId());
     }
 
