--- a/net/minecraft/world/level/block/state/properties/IntegerProperty.java
+++ b/net/minecraft/world/level/block/state/properties/IntegerProperty.java
@@ -7,8 +_,8 @@
 
 public final class IntegerProperty extends Property<Integer> {
     private final IntImmutableList values;
-    public final int min;
-    public final int max;
+    private final int min;
+    private final int max;
 
     private IntegerProperty(String name, int min, int max) {
         super(name, Integer.class);
