--- a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -4,7 +_,6 @@
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.codecs.RecordCodecBuilder;
-import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import io.netty.buffer.ByteBuf;
 import java.util.Arrays;
 import java.util.Iterator;
@@ -36,12 +_,12 @@
 import net.minecraft.world.level.block.BeehiveBlock;
 import net.minecraft.world.level.block.CampfireBlock;
 import net.minecraft.world.level.block.FireBlock;
-import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.gameevent.GameEvent;
 import org.slf4j.Logger;
+import org.teneted.neotenet.injection.world.level.block.entity.InjectionBeehiveBlockEntity;
 
-public class BeehiveBlockEntity extends BlockEntity {
+public class BeehiveBlockEntity extends BlockEntity implements InjectionBeehiveBlockEntity {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final String TAG_FLOWER_POS = "flower_pos";
     private static final String BEES = "bees";
@@ -84,6 +_,7 @@
     private final List<BeehiveBlockEntity.BeeData> stored = Lists.newArrayList();
     @Nullable
     public BlockPos savedFlowerPos;
+    public int maxBees = 3; // CraftBukkit - allow setting max amount of bees a hive can hold
 
     public BeehiveBlockEntity(BlockPos p_155134_, BlockState p_155135_) {
         super(BlockEntityType.BEEHIVE, p_155134_, p_155135_);
@@ -117,7 +_,7 @@
     }
 
     public boolean isFull() {
-        return this.stored.size() == 3;
+        return this.stored.size() == 3; // CraftBukkit
     }
 
     public void emptyAllLivingFromHive(@Nullable Player p_58749_, BlockState p_58750_, BeehiveBlockEntity.BeeReleaseStatus p_58751_) {
@@ -367,6 +_,18 @@
 
         public boolean hasNectar() {
             return this.occupant.entityData.getUnsafe().getBoolean("HasNectar");
+        }
+
+        public int getTicksInHive() {
+            return ticksInHive;
+        }
+
+        public void setTicksInHive(int ticksInHive) {
+            this.ticksInHive = ticksInHive;
+        }
+
+        public Occupant getOccupant() {
+            return occupant;
         }
     }
 
