--- a/net/minecraft/world/level/block/entity/BellBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BellBlockEntity.java
@@ -32,8 +_,8 @@
     public boolean shaking;
     public Direction clickDirection;
     private List<LivingEntity> nearbyEntities;
-    public boolean resonating;
-    public int resonationTicks;
+    private boolean resonating;
+    private int resonationTicks;
 
     public BellBlockEntity(BlockPos worldPosition, BlockState blockState) {
         super(BlockEntityTypes.BELL, worldPosition, blockState);
