--- a/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
@@ -43,8 +_,8 @@
     private static final int BURN_COOL_SPEED = 2;
     private static final int NUM_SLOTS = 4;
     private final NonNullList<ItemStack> items = NonNullList.withSize(4, ItemStack.EMPTY);
-    private final int[] cookingProgress = new int[4];
-    private final int[] cookingTime = new int[4];
+    public final int[] cookingProgress = new int[4];
+    public final int[] cookingTime = new int[4];
 
     public CampfireBlockEntity(BlockPos worldPosition, BlockState blockState) {
         super(BlockEntityTypes.CAMPFIRE, worldPosition, blockState);
