--- a/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -44,8 +_,8 @@
     private float activeRotation;
     private boolean isActive;
     private boolean isHunting;
-    public final List<BlockPos> effectBlocks = Lists.newArrayList();
-    public @Nullable EntityReference<LivingEntity> destroyTarget;
+    private final List<BlockPos> effectBlocks = Lists.newArrayList();
+    private @Nullable EntityReference<LivingEntity> destroyTarget;
     private long nextAmbientSoundActivation;
 
     public ConduitBlockEntity(BlockPos worldPosition, BlockState blockState) {
@@ -150,10 +_,8 @@
                         BlockPos testPos = worldPosition.offset(ox, oy, oz);
                         BlockState testBlock = level.getBlockState(testPos);
 
-                        for (Block type : VALID_BLOCKS) {
-                            if (testBlock.is(type)) {
-                                effectBlocks.add(testPos);
-                            }
+                        if (testBlock.isConduitFrame(level, testPos, worldPosition)) {
+                            effectBlocks.add(testPos);
                         }
                     }
                 }
@@ -218,7 +_,7 @@
         return candidates.isEmpty() ? null : EntityReference.of(Util.getRandom(candidates, level.getRandom()));
     }
 
-    public static AABB getDestroyRangeAABB(BlockPos worldPosition) {
+    private static AABB getDestroyRangeAABB(BlockPos worldPosition) {
         return new AABB(worldPosition).inflate(8.0);
     }
 
