--- a/net/minecraft/world/level/block/entity/LecternBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/LecternBlockEntity.java
@@ -35,7 +_,7 @@
     public static final int NUM_DATA = 1;
     public static final int SLOT_BOOK = 0;
     public static final int NUM_SLOTS = 1;
-    public final Container bookAccess = new Container() {
+    private final Container bookAccess = new Container() {
         @Override
         public int getContainerSize() {
             return 1;
@@ -156,7 +_,7 @@
         this.setChanged();
     }
 
-    public void setPage(int page) {
+    private void setPage(int page) {
         int newPage = Mth.clamp(page, 0, this.pageCount - 1);
         if (newPage != this.page) {
             this.page = newPage;
