--- a/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
@@ -59,11 +_,13 @@
         final SculkSpreader sculkSpreader;
         private final BlockState blockState;
         private final PositionSource positionSource;
+        public Level level; // CraftBukkit
 
         public CatalystListener(BlockState p_283224_, PositionSource p_283095_) {
             this.blockState = p_283224_;
             this.positionSource = p_283095_;
             this.sculkSpreader = SculkSpreader.createLevelSpreader();
+            this.sculkSpreader.level = level; // CraftBukkit
         }
 
         @Override
