--- a/net/minecraft/world/entity/decoration/ArmorStand.java
+++ b/net/minecraft/world/entity/decoration/ArmorStand.java
@@ -87,7 +_,7 @@
     private static final boolean DEFAULT_MARKER = false;
     private boolean invisible = false;
     public long lastHit;
-    private int disabledSlots = 0;
+    public int disabledSlots = 0;
 
     public ArmorStand(EntityType<? extends ArmorStand> type, Level level) {
         super(type, level);
@@ -479,7 +_,7 @@
         return this.isMarker();
     }
 
-    private void setSmall(boolean value) {
+    public void setSmall(boolean value) {
         this.entityData.set(DATA_CLIENT_FLAGS, this.setBit(this.entityData.get(DATA_CLIENT_FLAGS), 1, value));
     }
 
@@ -503,7 +_,7 @@
         return (this.entityData.get(DATA_CLIENT_FLAGS) & 8) == 0;
     }
 
-    private void setMarker(boolean value) {
+    public void setMarker(boolean value) {
         this.entityData.set(DATA_CLIENT_FLAGS, this.setBit(this.entityData.get(DATA_CLIENT_FLAGS), 16, value));
     }
 
