--- a/net/minecraft/world/entity/decoration/HangingEntity.java
+++ b/net/minecraft/world/entity/decoration/HangingEntity.java
@@ -57,7 +_,7 @@
         this.entityData.set(DATA_DIRECTION, direction);
     }
 
-    public void setDirection(Direction direction) {
+    protected void setDirection(Direction direction) {
         Objects.requireNonNull(direction);
         Validate.isTrue(direction.getAxis().isHorizontal());
         this.setDirectionRaw(direction);
@@ -84,7 +_,7 @@
             return false;
         }
 
-        boolean isSupported = BlockPos.betweenClosedStream(this.calculateSupportBox()).allMatch(pos -> {
+        boolean isSupported = BlockPos.betweenClosedStream(this.calculateSupportBox()).filter(pos -> !net.minecraft.world.level.block.Block.canSupportCenter(this.level(), pos, this.getDirection())).allMatch(pos -> {
             BlockState state = this.level().getBlockState(pos);
             return state.isSolid() || DiodeBlock.isDiode(state);
         });
