--- a/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -43,8 +_,8 @@
 import org.jspecify.annotations.Nullable;
 
 public class ItemFrame extends HangingEntity {
-    public static final EntityDataAccessor<ItemStack> DATA_ITEM = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.ITEM_STACK);
-    public static final EntityDataAccessor<Integer> DATA_ROTATION = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.INT);
+    private static final EntityDataAccessor<ItemStack> DATA_ITEM = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.ITEM_STACK);
+    private static final EntityDataAccessor<Integer> DATA_ROTATION = SynchedEntityData.defineId(ItemFrame.class, EntityDataSerializers.INT);
     public static final int NUM_ROTATIONS = 8;
     private static final float DEPTH = 0.0625F;
     private static final float WIDTH = 0.75F;
@@ -53,8 +_,8 @@
     private static final float DEFAULT_DROP_CHANCE = 1.0F;
     private static final boolean DEFAULT_INVISIBLE = false;
     private static final boolean DEFAULT_FIXED = false;
-    public float dropChance = 1.0F;
-    public boolean fixed = false;
+    private float dropChance = 1.0F;
+    private boolean fixed = false;
 
     public ItemFrame(EntityType<? extends ItemFrame> type, Level level) {
         super(type, level);
@@ -79,7 +_,7 @@
     }
 
     @Override
-    public void setDirection(Direction direction) {
+    protected void setDirection(Direction direction) {
         Objects.requireNonNull(direction);
         super.setDirectionRaw(direction);
         if (direction.getAxis().isHorizontal()) {
@@ -248,7 +_,7 @@
     private void removeFramedMap(ItemStack itemStack) {
         MapId mapId = this.getFramedMapId(itemStack);
         if (mapId != null) {
-            MapItemSavedData mapItemSavedData = MapItem.getSavedData(mapId, this.level());
+            MapItemSavedData mapItemSavedData = MapItem.getSavedData(itemStack, this.level());
             if (mapItemSavedData != null) {
                 mapItemSavedData.removedFromFrame(this.pos, this.getId());
             }
