--- a/net/minecraft/world/level/chunk/storage/EntityStorage.java
+++ b/net/minecraft/world/level/chunk/storage/EntityStorage.java
@@ -31,10 +_,10 @@
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final String ENTITIES_TAG = "Entities";
     private static final String POSITION_TAG = "Position";
-    public final ServerLevel level;
+    private final ServerLevel level;
     private final SimpleRegionStorage simpleRegionStorage;
     private final LongSet emptyChunks = new LongOpenHashSet();
-    public final ConsecutiveExecutor entityDeserializerQueue;
+    private final ConsecutiveExecutor entityDeserializerQueue;
 
     public EntityStorage(SimpleRegionStorage simpleRegionStorage, ServerLevel level, Executor mainThreadExecutor) {
         this.simpleRegionStorage = simpleRegionStorage;
@@ -94,9 +_,13 @@
                 ListTag entities = new ListTag();
                 chunk.getEntities().forEach(e -> {
                     TagValueOutput output = TagValueOutput.createWithContext(reporter.forChild(e.problemPath()), e.registryAccess());
+                    try {
                     if (e.save(output)) {
                         CompoundTag result = output.buildResult();
                         entities.add(result);
+                    }
+                    } catch (Exception ex) {
+                        LOGGER.error("An Entity type {} has thrown an exception trying to write state. It will not persist. Report this to the mod author", e.getType(), ex);
                     }
                 });
                 CompoundTag chunkTag = NbtUtils.addCurrentDataVersion(new CompoundTag());
