--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -13,8 +_,9 @@
 import net.minecraft.nbt.StreamTagVisitor;
 import net.minecraft.util.ExceptionCollector;
 import net.minecraft.world.level.ChunkPos;
+import org.celestial_artistry.neotenet.injection.world.level.chunk.storage.InjectionRegionFileStorage;
 
-public final class RegionFileStorage implements AutoCloseable {
+public final class RegionFileStorage implements AutoCloseable, InjectionRegionFileStorage {
     public static final String ANVIL_EXTENSION = ".mca";
     private static final int MAX_CACHE_SIZE = 256;
     public final Long2ObjectLinkedOpenHashMap<RegionFile> regionCache = new Long2ObjectLinkedOpenHashMap<>();
