--- a/net/minecraft/world/entity/monster/Creeper.java
+++ b/net/minecraft/world/entity/monster/Creeper.java
@@ -51,9 +_,9 @@
     private static final short DEFAULT_MAX_SWELL = 30;
     private static final byte DEFAULT_EXPLOSION_RADIUS = 3;
     private int oldSwell;
-    public int swell;
-    public int maxSwell = 30;
-    public int explosionRadius = 3;
+    private int swell;
+    private int maxSwell = 30;
+    private int explosionRadius = 3;
     private boolean droppedSkulls;
 
     public Creeper(EntityType<? extends Creeper> type, Level level) {
@@ -228,7 +_,7 @@
         }
     }
 
-    public void explodeCreeper() {
+    private void explodeCreeper() {
         if (this.level() instanceof ServerLevel level) {
             float explosionMultiplier = this.isPowered() ? 2.0F : 1.0F;
             this.dead = true;
