--- a/net/minecraft/server/level/DistanceManager.java
+++ b/net/minecraft/server/level/DistanceManager.java
@@ -44,7 +_,7 @@
     private final ThrottlingChunkTaskDispatcher ticketDispatcher;
     private final LongSet ticketsToRelease = new LongOpenHashSet();
     private final Executor mainThreadExecutor;
-    public int simulationDistance = 10;
+    private int simulationDistance = 10;
 
     protected DistanceManager(TicketStorage ticketStorage, Executor executor, Executor mainThreadExecutor) {
         this.ticketStorage = ticketStorage;
